### ISSUE TITLE
vscode.diff command with custom content provider is broken in 1.22

### ISSUE THREAD

**Issue by ArtemGovorov** [Microsoft/vscode#47271]
- VSCode Version: 1.22
- OS Version: macOS 10.13.3

Steps to Reproduce:

We are using the code below to display side-by-side diff from our extension:

```js
vscode.workspace.registerTextDocumentContentProvider('wallaby-diff', {
        provideTextDocumentContent(uri) {
          return new Buffer(uri.query.toString(), 'base64').toString();
        }
});

const showDiffCommand = vscode.Uri.parse('command:vscode.diff?' 
 + JSON.stringify([
  vscode.Uri.parse('wallaby-diff:data?' + new Buffer('value 1').toString('base64')),
  vscode.Uri.parse('wallaby-diff:data?' + new Buffer('value 2').toString('base64')),
  'Title'
]));

// using showDiffCommand in output and hover markdown
// like: `[Show side-by-side diff](${showDiffCommand})`
```

It works ok in 1.21, but is broken in 1.22 with the following message:

```
Unable to open 'vscode.diff': no provider for command:vscode.diff?%5B%7B%22%24mid%22%3A1%2C%22fsPath%22%3A%22data%22%2C%22external%22%3A%22wallaby-diff%3Adata...
```

Does this issue occur when all extensions are disabled?: Yes

**Comment by ArtemGovorov** [comment]
It looks like for some reason `vscode.diff` [can't be resolved as a command here](https://github.com/Microsoft/vscode/blob/392979d217dca43e833a1f86720740bbc3fbfd5f/src/vs/platform/opener/browser/openerService.ts#L47). However `_workbench.diff` works:

```js
const showDiffCommand = vscode.Uri.parse('command:_workbench.diff?' 
 + JSON.stringify([[
  vscode.Uri.parse('wallaby-diff:data?' + new Buffer('value 1').toString('base64')),
  vscode.Uri.parse('wallaby-diff:data?' + new Buffer('value 2').toString('base64')),
  'Title'
]]));
```

**Comment by jrieken** [comment]
@ArtemGovorov Just to double check, this only happens when using 'vscode.diff' from within a markdown link or, more generally speaking, when it is invoked from within the renderer, not the extension host, right?

@alexandrudima I am pretty sure this is a [regression from us keeping those commands private](https://github.com//Microsoft/vscode/commit/99b5cb6c9c791e28470c438fd58fa68443c00ade) to the extension host.

**Comment by ArtemGovorov** [comment]
@jrieken Correct. When executing the command via `vscode.commands.executeCommand('vscode.diff', ...` from the extension host, it's working as expected.

**Comment by jrieken** [comment]
Closed in `release/1.22`, needs a different fix in master
